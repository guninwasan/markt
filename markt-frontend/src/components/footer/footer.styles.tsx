import styled from "styled-components";
import { colors } from "../../utils";

const FooterContainer = styled.footer`
  background-color: ${colors.darkerHoverPrimary};
  color: white;
  padding: 5rem 0 10rem 0;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const MarktText = styled.div`
  font-size: 24px;
  margin-bottom: 3rem;
  color: ${colors.darkGrey};
`;

const LinksContainer = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 3rem;
`;

const Link = styled.a`
  color: white;
  margin: 0 20px;
  text-decoration: none;
  font-size: 16px;

  &:hover {
    text-decoration: underline;
  }
`;

const RightsReserved = styled.div`
  font-size: 14px;
  margin-bottom: 3rem;
`;

export { MarktText, FooterContainer, LinksContainer, Link, RightsReserved };
