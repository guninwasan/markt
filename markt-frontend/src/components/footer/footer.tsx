import React from "react";
import {
  FooterContainer,
  LinksContainer,
  FooterLink,
  RightsReserved,
  MarktText,
} from "./footer.styles";

const Footer = () => {
  return (
    <FooterContainer>
      <MarktText>
        MARKT by <i>Mizzica</i>
      </MarktText>
      <nav aria-label="Footer navigation">
        <LinksContainer>
          <FooterLink href="/about">About</FooterLink>
          <FooterLink href="/privacy-policy">Privacy Policy</FooterLink>
          <FooterLink href="/terms-of-use">Terms of Use</FooterLink>
          <FooterLink href="/support">Support / Report an issue</FooterLink>
        </LinksContainer>
      </nav>
      <RightsReserved>Â© 2024 All rights reserved.</RightsReserved>
    </FooterContainer>
  );
};

export { Footer };
