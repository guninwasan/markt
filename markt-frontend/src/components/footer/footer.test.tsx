import React from "react";
import { render, screen } from "@testing-library/react";
import { Footer } from "./footer";

describe("Footer", () => {
  it("should render the footer", () => {
    render(<Footer />);
    expect(screen.getByText(/MARKT by/i)).toBeInTheDocument();
  });

  it("should render all footer links", () => {
    render(<Footer />);
    expect(screen.getByText(/About/i)).toBeInTheDocument();
    expect(screen.getByText(/Privacy Policy/i)).toBeInTheDocument();
    expect(screen.getByText(/Terms of Use/i)).toBeInTheDocument();
    expect(screen.getByText(/Support \/ Report an issue/i)).toBeInTheDocument();
  });

  it("should render the rights reserved text", () => {
    render(<Footer />);
    expect(
      screen.getByText(/Â© 2024 All rights reserved./i)
    ).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
