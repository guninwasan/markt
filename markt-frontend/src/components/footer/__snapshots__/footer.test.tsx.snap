// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match the snapshot 1`] = `
<div>
  <footer
    class="sc-blHHSb hoxMpV"
  >
    <p
      class="sc-gtLWhw dDTpkr"
    >
      MARKT by 
      <i>
        Mizzica
      </i>
    </p>
    <nav
      aria-label="Footer navigation"
    >
      <div
        class="sc-egkSDF bdtBCy"
      >
        <a
          class="sc-fAUdSK cGNQhO"
          href="/about"
        >
          About
        </a>
        <a
          class="sc-fAUdSK cGNQhO"
          href="/privacy-policy"
        >
          Privacy Policy
        </a>
        <a
          class="sc-fAUdSK cGNQhO"
          href="/terms-of-use"
        >
          Terms of Use
        </a>
        <a
          class="sc-fAUdSK cGNQhO"
          href="/support"
        >
          Support / Report an issue
        </a>
      </div>
    </nav>
    <p
      class="sc-dntaoT gAQEaZ"
    >
      Â© 2024 All rights reserved.
    </p>
  </footer>
</div>
`;
